### PR TITLE
feature -> develop | vaahtwo permission page

### PR DESCRIPTION
**Pull request by wri-tanmoy-d001**
### [Task #209 - 2022-12-27 [Backend] > VaahCms > Roles > Docs > Development](https://team.webreinvent.com/backend/collab/project/vaahcms/tasks/209)

Time invested: hh:mm format | Billable: 05:00 | Non-Billable: hh:mm

Task Video Proof :- https://img-v4.getdemo.dev/screenshot/chrome_iCmSObrWJf.mp4

### CODE REVIEW

**Comment by themodernpk** [comment]
@wri-tanmoy-d001 https://drive.google.com/file/d/1o_J92STPbuk5i-0JnxgDINqv5zRsM_se/view `total_roles` and `total_users` should be the part of the `data`

Refer to https://docs.vaah.dev/guide/code.html#http-request-json-response 


**Response** can have only the following variables:
```json
{
	"success": true,
	"data": [],
	"errors": [],
	"messages": [],
	"hint": [],
	"code": 200,
	"meta": {}
}
```

**Comment by wri-tanmoy-d001** [comment]
@themodernpk SIR, 
I'll correct the API response structure and update the PR

File: docs/vaahcms-2/backend/permissions.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+parameter = [
+    'filter' => [
+        'q'            // for search queary (optionl)
+        'trashed'      // for include or exclude trashed data (optionl)
+        'is_active'    // for fetching only active or inactive data (optionl)
+        'sort'         // for sorting the data (optionl)
+    ];  
+];
+```
+
```
Comment (themodernpk) [inline docs/vaahcms-2/backend/permissions.md:217]: @wri-tanmoy-d001 @we-vikram-wri231 write `axios` sample code for reference

File: docs/vaahcms-2/backend/permissions.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+##### Sample Response
+
+```json
+{
+  "data": {
+      .............
```
Comment (themodernpk) [inline docs/vaahcms-2/backend/permissions.md:223]: @wri-tanmoy-d001 use no more than 3 dots `...`

File: docs/vaahcms-2/backend/permissions.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+      .............
+  },
+  "success": "true"
```
Comment (themodernpk) [inline docs/vaahcms-2/backend/permissions.md:226]: @wri-tanmoy-d001 `true` should not be in quotes

File: docs/vaahcms-2/backend/permissions.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+}
+```
+---
+
+#### Delete Permission
+
+###### Method: `DELETE`
+###### Action: `deleteItem`
```
Comment (themodernpk) [inline docs/vaahcms-2/backend/permissions.md:234]: @wri-tanmoy-d001 what is this `deleteItem`? is it from Vue or Model?

**Review by themodernpk — CHANGES_REQUESTED** [review]

File: docs/vaahcms-2/backend/permissions.md (hunk 4 above)
Comment (wri-tanmoy-d001) [reply docs/vaahcms-2/backend/permissions.md:234]: @themodernpk sir,

Sir, I'll update the PR
and sir `deleteItem` method part of the laravel model

File: docs/vaahcms-2/backend/permissions.md [hunk 5]
```diff
@@ -201,9 +201,34 @@ We mention some methods bellow which help you to understand the structure.
...
+
+```js
+
+async getList() {
+    let options = {
+        query: vaah().clone(this.query)
+    };
+    await vaah().ajax(
+        this.ajax_url,
+        this.afterGetList,
```
Comment (themodernpk) [inline docs/vaahcms-2/backend/permissions.md:217]: @wri-tanmoy-d001 `after` should be `suffix` not a `prefix`

**Review by themodernpk — CHANGES_REQUESTED** [review]

File: docs/vaahcms-2/backend/permissions.md [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+###### Method: `DELETE`
+###### Action: `PermissionsController@deleteItem`
+###### URL: `<public-url>/api/vaah/permissions/{id}`
+
+##### Sample Response
+
+```json
+{
+  "data": {},
+  "message" : [
```
Comment (themodernpk) [inline docs/vaahcms-2/backend/permissions.md:266]: @wri-tanmoy-d001 it's `messages` not `message`, it's plural.

**Review by themodernpk — CHANGES_REQUESTED** [review]

File: docs/vaahcms-2/backend/permissions.md (hunk 5 above)
Comment (wri-tanmoy-d001) [reply docs/vaahcms-2/backend/permissions.md:217]: Ok sir I'll update it

**Pull request merged**